### PR TITLE
Move typescript interfaces to PHC

### PR DESCRIPTION
**Pull request by tonidero**
This PR moves most of the typescript interfaces from the react native SDK to PHC. The interfaces are mostly a copy-paste of those in react-native-purchases. I haven't copied the `Purchases` interface yet. Will do that separately since that will be a bit trickier considering that in react native we are using static functions.

Counterpart react native PR: https://github.com/RevenueCat/react-native-purchases/pull/684

##### TODO
- [x] Make sure they will be usable by both react native and capacitor. This includes using them within the hybrids and making sure they are exposed to consumers. Confirmed these interfaces are visible in the react native PR
- [x] Make sure the documentation is generated correctly: While typedoc separates the documentation into an "external" visibility, we can make it show those by default, so it should still be ok.
- [x] Add typescript linter
- [x] Add typescript API tests (Added in #459)
- [ ] Move rest of interfaces to PHC (Missing purchases interface, will do that one separately)
- [x] Publish to NPM

### CODE REVIEW

File: typescript/tsconfig.json
Comment (tonidero) [inline typescript/tsconfig.json:1]: This is also a copy-paste of react-native with some slight differences. I was reading the documentation for each of these and most of them made sense to keep.

File: typescript/apitesters/customerInfo.ts
Comment (tonidero) [inline typescript/apitesters/customerInfo.ts:9]: About the API tests. I was thinking that, while we can have them here, we should also keep them in the hybrids just in case.

**Comment by tonidero** [comment]
Once I get the approval, I will deploy this package to npm

File: .circleci/config.yml
Comment (vegaro) [inline .circleci/config.yml:217]: Is this needed?

File: .circleci/config.yml
Comment (vegaro) [inline .circleci/config.yml:308]: same question here

File: typescript/package.json
Comment (vegaro) [inline typescript/package.json:2]: what about just?

```
@revenuecat/purchases-typescript
```

In case we want to add something that are not interfaces. The `hybrid` part I think is too generic and I wish I hadn't added it to `purchases-hybrid-common` lol

File: .circleci/config.yml
Comment (tonidero) [reply .circleci/config.yml:217]: Not really... I think I just need to update the `working_directory` to be a relative path instead. I just copied this from somewhere, but that's probably cleaner 👍

File: typescript/package.json
Comment (tonidero) [reply typescript/package.json:2]: Hmm I do wonder if it's too generic and might lead some people to confuse it and use it (note that this is meant to be internal... I guess we could call it `@revenuecat/purchases-typescript-internal` maybe? wdyt?

File: .circleci/config.yml
Comment (tonidero) [reply .circleci/config.yml:217]: So after testing it out, I actually think it's needed... If I change the working_directory but not this, it causes errors when creating the folders. I could cd to the folder before the command... But feels that this is better. Lmk if you think otherwise!

File: typescript/package.json
Comment (vegaro) [reply typescript/package.json]: I like `@revenuecat/purchases-typescript-internal` yes!

**Review by vegaro — APPROVED** [review]

**Pull request merged**
